File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <queue>
#include <algorithm>
using namespace std;
int n, m, v;
int visited1[1004], visited2[1004];
vector<int> adj[1004];

void dfs(int here) {
	visited1[here] = 1;
	cout << here << " ";
	for (int there : adj[here]) {
		if (visited1[there])continue;
		dfs(there);
	}
}

void bfs(int here) {
	queue<int> q;
	visited2[here] = 1;
	cout << here << " ";
	q.push(here);
	while (q.size()) {
		here = q.front();
		q.pop();
		for (int there : adj[here]) {
			if (visited2[there])continue;
			visited2[there] = visited2[here] + 1;
			cout << there << " ";
			q.push(there);
		}
	}
	
}

int main() {
	ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);

	cin >> n >> m >> v;
	for (int i = 0; i < m; i++) {
		int x, y;
		cin >> x >> y;
		adj[x].push_back(y);
		adj[y].push_back(x);
	}

	for (int i = 0; i <= n; i++) {
		sort(adj[i].begin(), adj[i].end());
	}

	dfs(v); 

	cout << '\n';

	bfs(v);

	return 0;
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.*;
import java.util.*;

public class Main{
    static StringBuilder sb = new StringBuilder();
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    static int n, m, v;
    static ArrayList<Integer>[] adj;
    static int[] visited1, visited2;
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        n = Integer.parseInt(st.nextToken());
        m = Integer.parseInt(st.nextToken());
        v = Integer.parseInt(st.nextToken());

        adj = new ArrayList[n + 4];
        for(int i = 0; i < n + 4; i++){
            adj[i] = new ArrayList<>();
        }

        visited1 = new int[n + 4];
        visited2 = new int[n + 4];

        for(int i = 0; i < m; i++){
            st = new StringTokenizer(br.readLine());

            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            adj[a].add(b);
            adj[b].add(a);
        }

        for(int i = 0; i < n + 4; i++){
            Collections.sort(adj[i]);
        }

        dfs(v);
        sb.append("\n");
        bfs(v);

        bw.write(sb + "");
        bw.flush();
        bw.close();
    }

    static void dfs(int here){
        visited1[here] = 1;
        sb.append(here + " ");

        for(int there: adj[here]){
            if(visited1[there] != 0)continue;
            dfs(there);
        }
    }

    static void bfs(int here){
        Queue<Integer> q = new ArrayDeque<>();

        visited2[here] = 1;
        sb.append(here + " ");

        q.offer(here);
        while(!q.isEmpty()){
            int tempHere = q.poll();

            for(int there: adj[tempHere]){
                if(visited2[there] != 0)continue;

                visited2[there] = visited2[tempHere] + 1;
                sb.append(there + " ");

                q.offer(there);
            }
        }
    }
}